### ISSUE TITLE
Fix display; remove stringify

### ISSUE THREAD

**Issue by martin-henz** [source-academy/js-slang#135]
Display should work "properly": handle circular data structures

stringify: no more in 2019 Source specs

**Comment by thomastanck** [comment]
Proposal

`display` will be the analogue of `x=>console.log(JSON.stringify(x);`, and add a new builtin named `console_log` which is the analogue of `x=>console.log(x);`. `display` should also handle circular data structures properly.

Thus `console_log` will use the interpreter toString to cast its args into a string whereas `display` will stringify instead.

The reasoning is to expose an easy to use function like `display` to display the contents of objects, array, etc., so we use stringify to display some valid JSON, which is mostly executable as javascript. On the other hand, we want to keep a lower level function like `console_log` so that if needed, users can display strings without having their quotes be escaped.

Examples:

`console_log('Bob says: "Hi Tom\'s assistant!"');` displays:
```
Bob says: "Hi Tom's assistant!"
```
`display('Bob says: "Hi Tom\'s assistant!"');` displays:
```
"Bob says: \"Hi Tom's assistant!\""
```

`console_log("Bob says: \"Hi Tom's assistant!\"");` displays:
```
Bob says: "Hi Tom's assistant!"
```
`display("Bob says: \"Hi Tom's assistant!\"");` displays:
```
"Bob says: \"Hi Tom's assistant!\""
```

**Comment by thomastanck** [comment]
Resolved by #165
